From: https://github.com/DependencyTrack/dependency-track/pull/5192
From: https://github.com/DependencyTrack/dependency-track/commit/ae8ddbfe6dfaaf573567abec7e8e3f852a117854
From ae8ddbfe6dfaaf573567abec7e8e3f852a117854 Mon Sep 17 00:00:00 2001
From: Arturo Borrero Gonzalez <arturo.borrero@chainguard.dev>
Date: Thu, 7 Aug 2025 13:18:15 +0200
Subject: [PATCH] CompressUtil: handle ArchiveException

Fixes this build problem:

[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /home/build/src/main/java/org/dependencytrack/util/CompressUtil.java:[43,96]
  unreported exception org.apache.commons.compress.archivers.ArchiveException; must be caught or declared to be thrown
[INFO] 1 error

Signed-off-by: Arturo Borrero Gonzalez <arturo.borrero@chainguard.dev>
---
 src/main/java/org/dependencytrack/util/CompressUtil.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/org/dependencytrack/util/CompressUtil.java b/src/main/java/org/dependencytrack/util/CompressUtil.java
--- a/src/main/java/org/dependencytrack/util/CompressUtil.java
+++ b/src/main/java/org/dependencytrack/util/CompressUtil.java
@@ -45,7 +45,7 @@ public static byte[] optionallyDecompress(final byte[] input) {
             if (ais.canReadEntryData(entry)) {
                 return IOUtils.toByteArray(ais);
             }
-        } catch (IOException e) {
+        } catch (IOException | org.apache.commons.compress.archivers.ArchiveException e) {
             // throw it away and return the original byte array
         }
         return input;
